api/router/correlation_test.go

```
@@ -0,0 +1,93 @@
+package router_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/allinbits/demeris-api-server/api/config"
+	"github.com/allinbits/demeris-api-server/api/database"
+	"github.com/allinbits/demeris-api-server/api/router"
+	"github.com/allinbits/emeris-utils/logging"
+	"github.com/allinbits/emeris-utils/store"
+	"github.com/cockroachdb/cockroach-go/v2/testserver"
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+type chainsResponse struct {
+	Chains []supportedChain `json:"chains"`
+}
+
+type supportedChain struct {
+	ChainName   string `json:"chain_name"`
+	DisplayName string `json:"display_name"`
+	Logo        string `json:"logo"`
+}
+
+func TestCorrelationIDMiddleWare(t *testing.T) {
+	r, cfg, observedLogs, tDown := setup(t)
+	defer tDown()
+	require.NotNil(t, r)
+
+	go r.Serve(cfg.ListenAddr)
+	time.Sleep(2 * time.Second)
+
+	client := http.Client{
+		Timeout: 2 * time.Second,
+	}
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s%s", cfg.ListenAddr, "/chains"), nil)
+	require.NoError(t, err)
+
+	id, _ := uuid.NewV4()
```

Why ignoring the err?